streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsKafkaClient.java

```diff
@@ -72,72 +72,88 @@
         }
 
-        public Config(Map<?, ?> originals) {
+        Config(Map<?, ?> originals) {
             super(CONFIG, originals, false);
         }
-    }
 
+    }
     private final KafkaClient kafkaClient;
     private final List<MetricsReporter> reporters;
     private final Config streamsConfig;
-
+    private final Map<String, String> defaultTopicConfigs = new HashMap<>();
     private static final int MAX_INFLIGHT_REQUESTS = 100;
 
-    public StreamsKafkaClient(final StreamsConfig streamsConfig) {
-        this(Config.fromStreamsConfig(streamsConfig));
-    }
 
-    public StreamsKafkaClient(final Config streamsConfig) {
+    StreamsKafkaClient(final Config streamsConfig,
+                       final KafkaClient kafkaClient,
+                       final List<MetricsReporter> reporters) {
         this.streamsConfig = streamsConfig;
+        this.kafkaClient = kafkaClient;
+        this.reporters = reporters;
+        extractDefaultTopicConfigs(streamsConfig.originalsWithPrefix(StreamsConfig.TOPIC_PREFIX));
+    }
 
+    private void extractDefaultTopicConfigs(final Map<String, Object> configs) {
+        for (final Map.Entry<String, Object> entry : configs.entrySet()) {
+            if (entry.getValue() != null) {
+                defaultTopicConfigs.put(entry.getKey(), entry.getValue().toString());
+            }
+        }
+    }
+
+
+    public static StreamsKafkaClient create(final Config streamsConfig) {
         final Time time = new SystemTime();
 
         final Map<String, String> metricTags = new LinkedHashMap<>();
         metricTags.put("client-id", StreamsConfig.CLIENT_ID_CONFIG);
 
         final Metadata metadata = new Metadata(streamsConfig.getLong(
-            StreamsConfig.RETRY_BACKOFF_MS_CONFIG),
-            streamsConfig.getLong(StreamsConfig.METADATA_MAX_AGE_CONFIG),
-            false
-        );
+                StreamsConfig.RETRY_BACKOFF_MS_CONFIG),
+                                               streamsConfig.getLong(StreamsConfig.METADATA_MAX_AGE_CONFIG), false);
         final List<InetSocketAddress> addresses = ClientUtils.parseAndValidateAddresses(streamsConfig.getList(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG));
         metadata.update(Cluster.bootstrap(addresses), Collections.<String>emptySet(), time.milliseconds());
 
         final MetricConfig metricConfig = new MetricConfig().samples(streamsConfig.getInt(CommonClientConfigs.METRICS_NUM_SAMPLES_CONFIG))
                 .timeWindow(streamsConfig.getLong(CommonClientConfigs.METRICS_SAMPLE_WINDOW_MS_CONFIG), TimeUnit.MILLISECONDS)
                 .tags(metricTags);
-        reporters = streamsConfig.getConfiguredInstances(ProducerConfig.METRIC_REPORTER_CLASSES_CONFIG,
-                MetricsReporter.class);
+        final List<MetricsReporter> reporters = streamsConfig.getConfiguredInstances(ProducerConfig.METRIC_REPORTER_CLASSES_CONFIG,
+                                                         MetricsReporter.class);
```

Comment:
Line 127: this is not introduced in this PR, but thinking about the prefix a bit more, we should use `KafkaAdminClient.JMX_PREFIX` as well since otherwise they will not be in the same mbean family.